src/features/device-state/routes/state.ts

```diff
@@ -362,24 +483,60 @@ const getConfig = (device: AnyObject) => {
 const getUserAppForState = (
 	device: AnyObject,
 	config: Dictionary<string>,
-): Partial<App> => {
+): AppV3 => {
 	const userAppFromApi: AnyObject = device.belongs_to__application[0];
 
 	// get the release of the main app that this device should run...
 	const release = getReleaseForDevice(device);
 
 	// grab the main app for this device...
 	return release == null
-		? {
+		? ({
 				name: userAppFromApi.app_name,
+				appId: userAppFromApi.id,
+				uuid: userAppFromApi.uuid,
 				services: {},
 				networks: {},
 				volumes: {},
-		  }
+		  } as AppV3)
 		: buildAppFromRelease(device, userAppFromApi, release, config);
 };
 
-const getDependent = (device: AnyObject): Dependent => {
+const getSupervisorAppForState = async (
+	device: any,
```

Comment:
This should at a minimum specify the attributes it requires:

```suggestion
	device: AnyObject & { should_be_managed_by__release: Array<{ belongs_to__application: ... }> }
```

Reply:
I changed to this AnyObject, but going beyond that did not feel like it made sense for this simple method which only touches a couple of properties and breaks with the existing pattern in this file.  If we want to start typing for use of the properties, I think it would be better in a refactoring PR to do so consistently.

Reply:
resolving since Akis is out.